test/unit/media/headset.test.ts

```diff
@@ -0,0 +1,159 @@
+import { Observable } from 'rxjs';
+import { SdkHeadset } from "../../../src/media/headset";
+import GenesysCloudWebrtcSdk from '../../../src/client';
+import GenesysCloudWebrtSdk from "../../../src";
+import HeadsetService, { ConsumedHeadsetEvents, VendorImplementation} from 'softphone-vendor-headsets';
+import { SimpleMockSdk } from '../../test-utils';
+
+let sdk: GenesysCloudWebrtSdk;
+let sdkHeadset: SdkHeadset;
+let headsetLibrary: HeadsetService;
+let headsetEvents$: Observable<ConsumedHeadsetEvents>;
+
+describe('SdkHeadset', () => {
+    beforeEach(() => {
+        jest.clearAllMocks();
+        headsetLibrary = HeadsetService.getInstance({ logger: console });
+        headsetEvents$ = headsetLibrary.headsetEvents$;
+        sdk = new SimpleMockSdk() as any;
+        sdkHeadset = new SdkHeadset(sdk);
+    })
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+    })
+
+    describe('constructor()', () => {
+        it('should start initialization', () => {
+            // const headset = new SdkHeadset(sdkMedia);
+            expect(sdkHeadset['sdk']).toBe(sdk);
+            expect(sdkHeadset['headsetLibrary']).toBe(headsetLibrary);
+            expect(sdkHeadset['headsetEvents$']).toStrictEqual(headsetEvents$);
+        })
+    })
+
+    describe('updateAudioInputDevice', () => {
+        it('should fetch the proper device and send it to the headset library', () => {
+            const testId = "testId";
+            const findCachedDeviceByIdAndKindSpy = jest.spyOn(sdk.media, 'findCachedDeviceByIdAndKind' as any).mockReturnValue({
```

Comment:
Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9082220)
```suggestion
            const findCachedDeviceByIdAndKindSpy = jest.spyOn(sdk.media, 'findCachedDeviceByIdAndKind' as unknown).mockReturnValue({
```